src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\Security;
+
+/**
+ * Object representation of a firewall configuration.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class FirewallConfig
+{
+    private $name;
+    private $requestMatcher;
+    private $securityEnabled;
+    private $stateless;
+    private $provider;
+    private $context;
+    private $entryPoint;
+    private $accessDeniedHandler;
+    private $accessDeniedUrl;
+    private $userChecker;
+    private $listeners;
+
+    public function __construct($name, $requestMatcher, $securityEnabled = true, $stateless = false, $provider = null, $context = null, $entryPoint = null, $accessDeniedHandler = null, $accessDeniedUrl = null, $userChecker = null, $listeners = array())
+    {
+        $this->name = $name;
+        $this->requestMatcher = $requestMatcher;
+        $this->securityEnabled = $securityEnabled;
+        $this->stateless = $stateless;
+        $this->provider = $provider;
+        $this->context = $context;
+        $this->entryPoint = $entryPoint;
+        $this->accessDeniedHandler = $accessDeniedHandler;
+        $this->accessDeniedUrl = $accessDeniedUrl;
+        $this->userChecker = $userChecker;
+        $this->listeners = $listeners;
+    }
+
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * Returns the request matcher service id.
+     *
+     * @return string
+     */
+    public function getRequestMatcher()
+    {
+        return $this->requestMatcher;
+    }
+
+    /**
+     * Returns either the security is enabled or not.
+     *
+     * @return bool
+     */
+    public function isSecurityEnabled()
+    {
+        return (bool) $this->securityEnabled;
+    }
+
+    /**
+     * Returns either anonymous is allowed or not.
+     *
+     * @return bool
+     */
+    public function allowsAnonymous()
+    {
+        return in_array('anonymous', $this->listeners);
```

Comment:
I'm not sure we should use strict comparison for strings.
After a quick look, it seems that both are used, so let's see wait for a third opinion :)